attiny84_sketch/pidog2/pidog2.ino

```diff
@@ -99,6 +99,16 @@ reg_t handleCommand(uint8_t cmd, reg_t indata) {
 
 void doSecondWork() {
     uint32_t s = rf.get(REG_STATUS);
+    bool pwr_state = s & _BV(STAT_PWR_ON);
```

Comment:
This is confusing enough that it probably deserves being pulled out and explained

```c
// Update and optionally clear the MCUSR register, which indicates *why* the
// attiny was reset. This is useful if you PiDog is resetting and you do not know
// why.
void read_or_clear_resetcause(uint32_t &s) {
    // Clear these status bits associated with the MCUSR register:
    s &= ~(_BV(STAT_WDRF) | _BV(STAT_BORF) | _BV(STAT_EXTRF) | _BV(STAT_PORF));
    // then either clear the MCUSR if requested or reload them from the MCUSR:
    if ( s & _BV(STAT_WDCLR)) { // rpi wants us to clear MCUSR
      MCUSR =0;
      s &= ~_BV(STAT_WDCLR);
    } else { // write the MCUSR low bits to the status register
      s |= uint32_t ((MCUSR & 0xF) << STAT_PORF);
    }
}
```

Then in the body of doSecondWork we can jus call `read_or_clear_resetcause(s)`.

Reply:
Done. Just need to test changes.

Reply:
Changes test okay.